airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -83,16 +83,22 @@ public void start(final WorkerSourceConfig workerSourceConfig, final Path jobRoo
          */
         isResetBasedForConfig = false;
       } else {
-        stateWrapper = StateMessageHelper.getTypedState(workerSourceConfig.getState().getState(), useStreamCapableState);
+        if (workerSourceConfig.getState() != null ) {
```

Comment:
Looks like there's a valid use case for a null state, can we add a unit test to make sure it's covered?

nit: trailing whitespace after null